### PR TITLE
[multistage]Fix chained literal functions evaluation

### PR DESCRIPTION
**Pull request by xiangfu0**
Fix: https://github.com/apache/pinot/issues/12245

### CODE REVIEW

**Review by walterddr — COMMENTED** [review]
looks like the integration test error is legit. can we take a look?

**Comment by xiangfu0** [comment]
> looks like the integration test error is legit. can we take a look?

yes, there is a missing check for cast and generate_array method catches it.

Will follow up another PR to support compile time all literal function evaluation when the return type is array.

**Review by walterddr — APPROVED** [review]

File: pinot-query-planner/src/test/resources/queries/LiteralEvaluationPlans.json [hunk 1]
```diff
@@ -160,6 +160,29 @@
...
+      },
+      {
+        "description": "filter with two functions chaining",
+        "sql": "EXPLAIN PLAN FOR SELECT count(*) FROM a WHERE col1 > ToEpochDays(fromDateTime('1970-01-15', 'yyyy-MM-dd'))",
+        "output": [
+          "Execution Plan",
+          "\nLogicalAggregate(group=[{}], agg#0=[COUNT($0)])",
+          "\n  PinotLogicalExchange(distribution=[hash])",
+          "\n    LogicalAggregate(group=[{}], agg#0=[COUNT()])",
+          "\n      LogicalFilter(condition=[>(CAST($0):BIGINT NOT NULL, 14)])",
```
Comment (walterddr) [inline pinot-query-planner/src/test/resources/queries/LiteralEvaluationPlans.json:181]: could you add another test case where CAST is not being involved (to validate the other codepath)

Comment (xiangfu0) [reply pinot-query-planner/src/test/resources/queries/LiteralEvaluationPlans.json:251]: done.

**Pull request merged**